### PR TITLE
fixed overwriting containerPort names issue

### PR DESCRIPTION
**Pull request by natasha41575**
#2767 

This issue will eventually be fixed at its root by schema, but for now this is a temporary workaround.

### CODE REVIEW

**Comment by Shell32-Natsu** [comment]
Shouldn't we fix this by customizing schema? Or this is a quick work around?

File: kyaml/yaml/walk/map.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ func (l Walker) walkMap() (*yaml.RNode, error) {
 
 	// recursively set the field values on the map
 	for _, key := range l.fieldNames() {
+		if key == "ports" {
```
Comment (Shell32-Natsu) [inline kyaml/yaml/walk/map.go:30]: Although this hack can bypass the port merging issue, but `yaml.walk` is a very fundamental function in kyaml and kyaml is not only used by kustomize. Blindly skip `ports` field may have unpredictable results in some cases.

Comment (natasha41575) [reply kyaml/yaml/walk/map.go:30]: Understood, I'll keep looking through the code over the next few days and try to fix it properly via schema and update the PR.

**Comment by monopole** [comment]
believe the choices are

1) special code in kustomize, like this pr, but agree with @Shell32-Natsu that we shouldn't put it in walk and i think 2) might be easier.

2) a script to modify swagger.json after it is captured but before swagger.go is generated

3) fix it upstream in the apiserver (file an issue, etc.)

To do (2) we add commands to https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/Makefile#L37

So, figure out the edit to make. If the edit is just replace, insert or delete a line, use sed.

If it requires anything more complex, use https://github.com/chrisdone/jl

It's nice that it doesn't require a program.

other choices here:  https://github.com/burningtree/awesome-json#command-line-tools

**Pull request closed without merging**